pyfr/integrators/dual/pseudo/base.py

```diff
@@ -62,9 +62,22 @@ def __init__(self, backend, systemcls, rallocs, mesh,
         self._subdims = [elementscls.convarmap[self.system.ndims].index(v)
                          for v in elementscls.dualcoeffs[self.system.ndims]]
 
-        # Pointwise kernels for integrator
-        self.ikernels = {}
-        self.intgkernels = defaultdict(proxylist)
+        # Convergence tolerances
+        self._pseudo_residtol = residtol = OrderedDict({})
+        for v in elementscls.convarmap[self.system.ndims]:
+            try:
+                residtol[v] = cfg.getfloat(sect,
```

Comment:
See if we can get this on one line.  Might be easier to use '...' + v here (usually it is a bad idea, but this use case is something of an exception as it is just appending).